jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/Dispatcher.java

```diff
@@ -271,6 +272,18 @@ public String getRequestURI()
         @Override
         public Object getAttribute(String name)
         {
+            //Servlet Spec 9.4.2 no forward attributes if a named dispatcher
+            if (_named != null && name != null && name.startsWith(__FORWARD_PREFIX))
+                return null;
+
+            //Servlet Spec 9.4.2 must return the values from the original request
+            if (name != null && name.startsWith(__FORWARD_PREFIX))
+            {
+                Object val = super.getAttribute(name);
+                if (val != null)
+                    return val;
+            }
```

Comment:
I'm a bit worried that this approach will get confused if the value is meant to be null (eg a null pathInfo).
So perhaps use the attribute technique to fetch the original request, each `ForwardRequest` asks it's wrapped request for the original request attribute. When it get's null, it knows it is the first forward and returns it's wrapper.